app/src/androidTest/java/com/github/se/endToEnd/SettingsFlowEndToEnd.kt

```diff
@@ -0,0 +1,293 @@
+package com.github.se.endToEnd
+
+import android.content.Intent
+import androidx.compose.ui.semantics.SemanticsProperties
+import androidx.compose.ui.semantics.SemanticsProperties.ToggleableState
+import androidx.compose.ui.semantics.getOrNull
+import androidx.compose.ui.test.assertHasClickAction
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.assertIsNotDisplayed
+import androidx.compose.ui.test.assertIsOff
+import androidx.compose.ui.test.assertIsOn
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onAllNodesWithTag
+import androidx.compose.ui.test.onFirst
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performScrollTo
+import androidx.compose.ui.test.performTextClearance
+import androidx.compose.ui.test.performTextInput
+import androidx.core.app.ActivityCompat
+import androidx.test.core.app.ActivityScenario
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.github.se.icebreakrr.MainActivity
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FirebaseFirestore
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import javax.inject.Inject
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@HiltAndroidTest
+class SettingsFlowEndToEnd {
+  private val intent =
+      Intent(InstrumentationRegistry.getInstrumentation().targetContext, MainActivity::class.java)
+
+  @Inject lateinit var authInjected: FirebaseAuth
+  @Inject lateinit var firestoreInjected: FirebaseFirestore
+  @Inject lateinit var authStateListener: FirebaseAuth.AuthStateListener
+
+  @get:Rule var hiltRule = HiltAndroidRule(this)
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setup() {
+    hiltRule.inject()
+    intent.putExtra("IS_TESTING", true)
+    ActivityCompat.setPermissionCompatDelegate(TestPermissionDelegate())
+  }
+
+  @Test
+  fun SettingsFlowEndToEndTest() {
+    ActivityScenario.launch<MainActivity>(intent).use { scenario ->
+      // Screen 1 : Around You Screen
+      composeTestRule.onNodeWithTag("aroundYouScreen").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("bottomNavigationMenu").assertIsDisplayed()
+      composeTestRule.waitForIdle()
+      composeTestRule.onAllNodesWithTag("profileCard").onFirst().assertIsDisplayed()
```

Comment:
This leads to the same result as what I wrote because in the two cases, the test will not pass if there are no profile cards. But I can make that change, I feel like it can be more robust .